pandora-common/src/character/restrictionsManager.ts

```diff
@@ -540,3 +569,21 @@ export class CharacterRestrictionsManager {
 		return { allowed: true };
 	}
 }
+
+export function ResolveAssetPreference({ preferences }: Immutable<{ preferences: AssetPreferences; }>, asset: Asset, _source?: CharacterId): AssetPreferenceType {
```

Comment:
Maybe this could receive only the preferences instead of wrapping them in an object and then immediately destructing?